packages/axelar-std-derive/src/lib.rs

```diff
@@ -219,3 +220,221 @@ pub fn derive_upgradable(input: TokenStream) -> TokenStream {
         }
     }.into()
 }
+
+/// Implements the Event and EventTestUtils traits for a Soroban contract event.
+///
+/// Fields without a `#[data]` attribute are used as topics, while fields with `#[data]` are used as event data.
+/// The event name can be specified with `#[event_name(...)]` or will default to the struct name in snake_case (minus "Event" suffix).
+///
+/// # Example
+/// ```rust
+/// # mod test {
+/// use core::fmt::Debug;
+/// use stellar_axelar_std::events::Event;
+/// use stellar_axelar_std_derive::IntoEvent;
+/// use soroban_sdk::{Address, contract, contractimpl, Env, String};
+///
+/// #[derive(Debug, PartialEq, IntoEvent)]
+/// pub struct ApproveEvent {
+///     pub owner: Address,
+///     pub spender: Address,
+///     #[data]
+///     pub amount: String,
+/// }
+///
+/// #[derive(Debug, PartialEq, IntoEvent)]
+/// #[event_name("transfer")]
+/// pub struct TransferEvent {
+///     pub from: Address,
+///     pub to: Address,
+///     #[data]
+///     pub amount: String,
+/// }
+///
+/// #[contract]
+/// pub struct TokenContract;
+///
+/// #[contractimpl]
+/// impl TokenContract {
```

Comment:
```suggestion
/// pub struct Token;
///
/// #[contractimpl]
/// impl Token {
```
contract suffix isn't really added in stellar